prog/dftb+/lib_dftbplus/mainapi.F90

```diff
@@ -104,6 +120,18 @@ subroutine getStressTensor(env, stress)
 
   end subroutine getStressTensor
 
+   subroutine getVirial(env, virial)
+
+    !> instance
+    type(TEnvironment), intent(inout) :: env
+
+    !> resulting gradients wrt atom positions
+    real(dp), intent(out) :: virial(:,:)
+
+    call recalcGeometry(env)
+    virial(:,:) = totalStress*CellVol
```

Comment:
Since the stress tensor is saved in DFTB+ and not the virial, it might not be available for molecular system where the stress tensor is ill-definied. A check like in the `getStressTensor` routine would be necessary here:

```suggestion
    if (.not. tStress) then
      call error("Virial not available, you must initialise your calculator with&
          & this property enabled.")
    end if

    call recalcGeometry(env)
    virial(:,:) = totalStress*CellVol
```